src/tool/abi/types.cpp

```diff
@@ -108,6 +149,12 @@ void enum_type::write_cpp_definition(writer& w) const
     {
         if (auto value = field.Constant())
         {
+            auto isExperimental = ::is_experimental(field);
+            if (isExperimental)
+            {
+                w.write("#if defined(ENABLE_WINRT_EXPERIMENTAL_TYPES)\n");
+            }
+
```

Comment:
might make a nice little experimental_guard raii

Reply:
Not a bad idea, but the write ordering isn't always such that destructor order is desired. Could add scopes to force it, but I'm not sure it's necessarily worth the hassle. And it would only be usable in four locations.